### DISCUSSION TITLE
Show virtual machines

### DISCUSSION THREAD

**Discussion by urbaman** [mattieserver/netbox-topology-views#258] [Q&A] [answered]
Hi,

I'm trying out the plugin, and it's wonderful.

Is there a way for it to show virtual machines (they're not shown in my topology, however I set the filters)?

Thank you!

**Comment by mattieserver** [comment, marked as answer]
Hi, 

No, we currently don't show virtual machines.

I might add this later but it's currently very low prio.

**Reply by JanCBredow** [reply]
uncool :c 

would love this too!

**Reply by dreng** [reply]
> uncool :c

If you think this is uncool, I suggest to let the NetBox Dev Team know about your thoughts. Maybe they change their minds and implement connections between virtual interfaces. Actually, we can't add features to this plugin if NetBox does not provide appropriate data.

**Comment by urbaman** [comment]
Well, I'll do without for the moment...

Thank you very much for your work!

**Comment by dreng** [comment]
I don't think we are able to implement this in a meaningful way. NetBox provides interfaces for virtual machines but doesn't provide any virtual cables. So the best thing we could do for now is to just show the virtual machines. But what sense do those machines make in a topology view if there's no connection between them?

**Reply by urbaman** [reply]
Well, having virtual cables to attach a VM to its hypervisor's port/bridge would be a great addition to Netbox, and I understand it's not that meaningful in a topology without that (could be similar to unconnected devices in a way...), but I'm actually missing a lot of devices/services on the network this way.

Anyway, it might be something to keep an eye on in the future ;)

Thank you again!

**Reply by dreng** [reply]
> Anyway, it might be something to keep an eye on in the future ;)

Definitely, yes. Unfortunately we are not NetBox developers. As far as I remember Jeremy Stretch once wrote it is not very likely that virtual cables (and switches) will be implemented.

**Reply by toha-tiger** [reply]
Maybe it's possible to build the links or at least grouping for VM's by using Clusters data?
The VM can be assigned to a Cluster and Devices are also bindable to Clusters.
